Trustroots/trustroots
# User: add API to retrieve avatar

#### Proposed Changes

* Adds an API endpoint to retrieve user avatar easily without frontend logic.

Currently does redirect to the final image, but next steps should likely be:
- Read and stream the image in the serverside instead
- Add caching layer so that we don't transfer gigabytes of images on the internet repeatedly for nothing
- Consider adding caching layer to ensure hit on DB when opening pages with multiple avatars (maybe do some measurements first)

We could probably test this with the mobile app (https://github.com/Trustroots/trustroots-mobile) first before doing so on the web.

#### Testing Instructions

* Upload avatar image to your user
* Open `http://localhost:3000/api/users/[USER_ID]/avatar`
* You should see your image
* Play with `source` and `size` URL args, you can see valid values from code

#### Known problems:
- Users with `public:false` return 404 instead of default avatar

Resolves #982
